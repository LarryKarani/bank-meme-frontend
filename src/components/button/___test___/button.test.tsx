import React from 'react';
import  ReactDOM from 'react-dom';
import Button from '../index';
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom';
import renderer from "react-test-renderer";

afterEach(cleanup)

it("renders without crashing", () => {
    const  div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div);
});

it("renders button correctly", () => {
    const {getByTestId} = render(<Button text="click me"></Button>)
    expect(getByTestId('button')).toHaveTextContent("click me")
})

it("renders button correctly", () => {
    const {getByTestId} = render(<Button text="save"></Button>)
    expect(getByTestId('button')).toHaveTextContent("save")
})

it("matches snapshot", () => {
    const tree = renderer.create(<Button text="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot()
})
