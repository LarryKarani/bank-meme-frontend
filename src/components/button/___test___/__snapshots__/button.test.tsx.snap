// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="ant-btn ant-btn-primary"
  data-testid="button"
  onClick={[Function]}
  type="button"
>
  <span>
    save
  </span>
</button>
`;
